taskcluster/async/asyncclient.py

```diff
@@ -137,14 +137,16 @@ class AsyncBaseClient(BaseClient):
                     superExc=rerr
                 )
 
-            # Handle non 2xx status code and retry if possible
-            try:
-                response.raise_for_status()
-                if response.status == 204:
-                    return None
+            status = response.status
+            if status == 204:
+                return None
 
-            except aiohttp.ClientError as rerr:
-                status = response.status
+            # Handle non 2xx status code and retry if possible
+            if status < 200 or status >= 300:
+                rerr = aiohttp.ClientResponseError(
```

Comment:
I'm not raising the exception, just handling it like it would look if we had caught `raise_for_status`, if `raise_for_status` also raised for <200 or >=300.  The alternative was to craft an error message string, and then I have no exception to set as `superExc` in the taskcluster exception.  I wasn't sure if anything downstream is assuming there is an exception set in `superExc`, otherwise I would have set the error string there.

So we're still raising taskcluster exceptions; we're just using `rerr` as the source of message in `log.warn`, and as the `superExc` when we're raising a taskcluster exception.

Do you still want changes here?